ds3-autogen-parser/src/main/java/com/spectralogic/ds3autogen/models/xml/typemap/TypeMap.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * ******************************************************************************
+ *   Copyright 2014-2017 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+package com.spectralogic.ds3autogen.models.xml.typemap;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+public class TypeMap {
+
+    @JsonProperty("TypeMap")
+    private ImmutableList<TypeMapElement> typeMap;
+
+    public ImmutableMap<String, String> toMap() {
+        final ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+        typeMap.forEach(element -> builder.put(element.getContractType(), element.getSdkType()));
+        return builder.build();
+    }
+
+    public void setTypeMap(ImmutableList<TypeMapElement> typeMap) {
```

Comment:
Why is the setter for this needed?  Why not pass it in via the constructor?

Reply:
json parsing class

Reply:
I guess I was confused since it has no getter